app.rb

```diff
@@ -0,0 +1,127 @@
+require './book'
+require './student'
+require './rental'
+require './teacher'
+
+class App
+  def initialize(parent)
+    @parent = parent
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def list_of_books
+    if @books.empty?
+      puts 'No Book In The List.'
+    else
+      puts 'List of Books available.'
+      @books.each { |book| puts "Title: #{book.title} By #{book.author}" }
+    end
+    @parent.show_menu
+  end
+
+  def list_of_people
+    if @people.empty?
+      puts 'List Of People is empty'
+    else
+      puts 'List of People:'
+      @people.each do |person|
+        puts "NAME: #{person.name}, CLASS: #{person.class}, ID: #{person.id}, AGE: #{person.age}"
+      end
+    end
+    @parent.show_menu
+  end
+
+  def create_person
+    puts 'Do you want to create a Student(1) or Teacher(2)? [ENTER NUMBER]:'
+    person_input = gets.chomp
+    case person_input
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Please add a valid input!'
+    end
+    @parent.show_menu
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
+    @people.push(Student.new(age, name, parent_permission: parent_permission))
+    puts
+    puts 'Student is successfully created'
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people.push(Teacher.new(age, specialization, name))
+    puts
+    puts 'Teacher is successfully created'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books.push(Book.new(title, author))
+    puts 'Book created successfully'
+    @parent.show_menu
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: '#{book.title}', Author: #{book.author}" }
+
+    puts 'Enter the number of the book:'
+    book_num = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
```

Comment:
- [ ] The same issue is present here ⚠👇 Kindly look into this so that we can provide a clear insight and a good user experience ✌

![image](https://github.com/Booze33/OOP-school-library/assets/106119105/27838dfa-ffe7-4893-ab50-356271942aab)